### PR TITLE
add Test::DistManifest to TEST_REQUIRES list

### PR DESCRIPTION
**Pull request by bor**
require after f108125e

### CODE REVIEW

**Comment by karenetheridge** [comment]
No, this module should not be required for normal user installations. You can see the test is guarded with this code:

```
BEGIN {
    plan skip_all => 'these tests are for authors only!'
        unless -d '.git' || $ENV{AUTHOR_TESTING};
}
```

File: t/distmanifest.t [hunk 1]
```diff
@@ -7,5 +7,7 @@ BEGIN {
         unless -d '.git' || $ENV{AUTHOR_TESTING};
 }
 
-use Test::DistManifest;
+eval { require Test::DistManifest; };    ## no critic (ErrorHandling::RequireCheckingReturnValueOfEval)
+plan skip_all => 'Test::DistManifest required' if $@;
```
Comment (karenetheridge) [inline t/distmanifest.t]: This is not what we want.  When a developer is releasing the code, we *need* this test to be run, to ensure the MANIFEST is up to date. Allowing the test to be skipped defeats the whole point.

**Comment by bor** [comment]
Then you should switch to 
``eval "use Test::DistManifest;"``
or 
``require Test::DistManifest;``

because `use` try to load module anyway (at compile time)

For example for me this test broken now. As I'm dont have $ENV{AUTHOR_TESTING} and the `Test::DistManifest` module installed.

**Comment by bor** [comment]
like this
4c9af718910f85c3f60d3a70ba58a175dce682ea
```
-use Test::DistManifest;
-manifest_ok();
+require Test::DistManifest;
+Test::DistManifest::manifest_ok();
```

**Comment by karenetheridge** [comment]
> because `use` try to load module anyway (at compile time)

No, if the conditions are met for the `skip_all`, the module will not be loaded. Try it and see!

The conditions are not met for you because of `-d '.git'`.

**Comment by bor** [comment]
Ah, the condition under BEGIN. My miss.
Sorry for the noise :)

**Pull request closed without merging**